File: backend/app/schemas/participant.py
from pydantic import BaseModel, Field
from typing import Optional


class JoinSessionRequest(BaseModel):
    display_name: str = Field(
        min_length=1,
        max_length=50
    )

    transport: str = Field(
        default="wifi"
    )


class ParticipantResponse(BaseModel):
    participant_id: str
    display_name: str
    room_code: str
    connection_state: str
    transport: str
    muted: bool
    volume: float

File: frontend/src/types/participant.ts
export type ConnectionType =
  | "wifi"
  | "bluetooth";

export type AudioTransportType =
  | "wifi"
  | "bluetooth";

export type ConnectionState =
  | "connecting"
  | "connected"
  | "disconnected"
  | "failed";

export type FloorState =
  | "none"
  | "requested"
  | "granted";

export type SpeakingMode =
  | "open"
  | "controlled";

export interface VMICParticipant {
  participantId: string;
  displayName: string;

  connectionType:
    ConnectionType;

  transport:
    AudioTransportType;

  connectionState:
    ConnectionState;

  volume: number;

  muted: boolean;

  audioLevel: number;

  speaking: boolean;

  floorState: FloorState;

  pushToTalkActive: boolean;
}
